### PR TITLE
Created the resource-router-mapping transform

### LINKED ISSUE #50: Convert calls to this.resource in the router map to this.route { resetNamespace: true }
See https://github.com/emberjs/ember.js/pull/11517 for more info on this change.

### PR DESCRIPTION
**Pull request by marcioj**
Fixes #50

### CODE REVIEW

**Comment by knownasilya** [comment]
Nice!

**Comment by abuiles** [comment]
@marcioj Thanks! I'll give it a whole review tomorrow morning and release :)

File: lib/formulas/resource-router-mapping.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    builders.identifier('resetNamespace'),
+    builders.literal(true)
+  );
+
+  callee.property.name = 'route';
+  optionsObject.properties.push(resetNamespaceOption);
+}
+
+function isRoute(route) {
+  return route.type === 'ExpressionStatement' &&
```
Comment (abuiles) [inline lib/formulas/resource-router-mapping.js]: @marcioj can you use recast's API to check the type ?

You can do something like

```
var types            = recast.types.namedTypes;

type. ExpressionStatement.check(route)
type.CallExpression.check(route.expression.callee)
```

Comment (marcioj) [inline lib/formulas/resource-router-mapping.js]: Good point, I was mixing the two forms. I updated to use the recast namedTypes though

**Comment by abuiles** [comment]
@marcioj thanks!

**Pull request merged**